File: CS/Q130069/Form1.cs
using DevExpress.Utils;
using DevExpress.XtraGrid.Views.BandedGrid;
using DevExpress.XtraGrid.Views.BandedGrid.ViewInfo;
using System;
using System.Data;
using System.Windows.Forms;

namespace Q130069
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void bandedGridView1_MouseDown(object sender, MouseEventArgs e)
        {
            if (Control.ModifierKeys != (Keys.Shift | Keys.Control)) return;
            BandedGridView view = (BandedGridView)sender;
            BandedGridHitInfo hInfo = view.CalcHitInfo(e.Location);
            if (hInfo.InColumn)
            {
                view.ClearSelection();
                SelectCells(hInfo.Column);
            }
            else if (hInfo.InBandPanel && hInfo.Band != null)
            {
                view.ClearSelection();
                SelectCells(hInfo.Band);
            }
            else
                return;
            ((DXMouseEventArgs)e).Handled = true;
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            bindingSource1.DataSource = GetPersonDataTable();

        }
        DataTable GetPersonDataTable()
        {
            DataTable table = new DataTable();
            table.TableName = "Persons";
            table.Columns.Add(new DataColumn("FirstName", typeof(string)));
            table.Columns.Add(new DataColumn("SecondName", typeof(string)));
            table.Columns.Add(new DataColumn("Age", typeof(int)));
            table.Columns.Add(new DataColumn("ID", typeof(int)));
            for (int i = 0; i < 50; i++)
                table.Rows.Add("FirstName " + i, "SecondName " + i, 20 + i / 2, i);
            return table;
        }

        private void SelectCells(BandedGridColumn column)
        {
            for (int i = 0; i < column.View.RowCount; i++)
                column.View.SelectCell(i, column);
        }

        private void SelectCells(GridBand band)
        {
            foreach (BandedGridColumn column in band.Columns)
                SelectCells(column);
        }
    }
}

File: VB/Q130069/Form1.vb
Imports DevExpress.Utils
Imports DevExpress.XtraGrid.Views.BandedGrid
Imports DevExpress.XtraGrid.Views.BandedGrid.ViewInfo
Imports System
Imports System.Data
Imports System.Windows.Forms


Partial Public Class Form1
    Inherits Form

    Public Sub New()
        InitializeComponent()
    End Sub

    Private Sub bandedGridView1_MouseDown(ByVal sender As Object, ByVal e As MouseEventArgs) Handles bandedGridView1.MouseDown
        If Control.ModifierKeys <> (Keys.Shift Or Keys.Control) Then Return
        Dim view As BandedGridView = CType(sender, BandedGridView)
        Dim hInfo As BandedGridHitInfo = view.CalcHitInfo(e.Location)

        If hInfo.InColumn Then
            view.ClearSelection()
            SelectCells(hInfo.Column)
        ElseIf hInfo.InBandPanel AndAlso hInfo.Band IsNot Nothing Then
            view.ClearSelection()
            SelectCells(hInfo.Band)
        Else
            Return
        End If

        CType(e, DXMouseEventArgs).Handled = True
    End Sub

    Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
        bindingSource1.DataSource = GetPersonDataTable()
    End Sub

    Private Function GetPersonDataTable() As DataTable
        Dim table As DataTable = New DataTable()
        table.TableName = "Persons"
        table.Columns.Add(New DataColumn("FirstName", GetType(String)))
        table.Columns.Add(New DataColumn("SecondName", GetType(String)))
        table.Columns.Add(New DataColumn("Age", GetType(Integer)))
        table.Columns.Add(New DataColumn("ID", GetType(Integer)))

        For i As Integer = 0 To 50 - 1
            table.Rows.Add("FirstName " & i, "SecondName " & i, 20 + i / 2, i)
        Next

        Return table
    End Function

    Private Sub SelectCells(ByVal column As BandedGridColumn)
        For i As Integer = 0 To column.View.RowCount - 1
            column.View.SelectCell(i, column)
        Next
    End Sub

    Private Sub SelectCells(ByVal band As GridBand)
        For Each column As BandedGridColumn In band.Columns
            SelectCells(column)
        Next
    End Sub
End Class
